File: test-harness/Cucumber/features/step_definitions/AdditionSteps.rb
Given(/^the calculator has clean memory$/) do
  @result = 0
  @numbersList = []
end

Given(/^I have entered (\d+) into the calculator$/) do |arg1|
  @numbersList << arg1.to_i
end

Given(/^I have entered (\d+)\.(\d+) into the calculator$/) do |arg1, arg2|
  expect("this is a hacky way of making the scenario with a non-integer number").to eql("fail")
end

When(/^I press add$/) do
  @numbersList.each do |number|
      @result = @result + number
  end
end

Then(/^the result should be (\d+) on the screen$/) do |arg1|
  expect(@result).to eql(arg1.to_i)
end

Then(/^the result should be (\d+)\.(\d+) on the screen$/) do |arg1, arg2|
  expect("this is a hacky way of making the scenario with a non-integer number").to eql("fail")
end

Given(/^the background step fails$/) do
  expect("true").to eql("false")
end

File: test-harness/CucumberJS/features/stepdefinitions/AdditionSteps.js
'use strict';

var Calculator = require(process.cwd() + '/script/model/calculator');
var assert = require('assert');
var {defineSupportCode} = require('cucumber');

defineSupportCode(function({Given, When, Then}) {
  var myCalculator;

  Given(/^the calculator has clean memory$/, function (callback) {
    myCalculator = Calculator.create();
    callback();
  });

  Given(/^I have entered (.*) into the calculator$/, function (firstNumber, callback) {
    myCalculator.enter(firstNumber);
    callback();
  });

  Given(/^I have entered (\d+)\.(\d+) into the calculator$/, function (firstNumber, secondNumber, callback) {
    assert.equal("this is a hacky way of making the scenario with a non-integer number", "fail");
    callback();
  });

  When(/^I press add$/, function (callback) {
     myCalculator.add();
     callback();
   });

  Then(/^the result should be (.*) on the screen$/, function (arg1, callback) {
    assert.equal(myCalculator.result, parseInt(arg1));
    callback();
  });

  Given(/^the background step fails$/, function (callback) {
    assert.equal("true", "false");
    callback();
  });
});
